src/proxy/inbound_passthrough.rs

```diff
@@ -55,7 +56,17 @@ impl InboundPassthrough {
                         }
                     });
                 }
-                Err(e) => error!("Failed TCP handshake {}", e),
+                Err(e) => {
+                    debug_assert_eq!(e.get_ref().unwrap().to_string(), ERR_TOKIO_RUNTIME_SHUTDOWN);
+                    if e.get_ref()
+                        .unwrap()
```

Comment:
unwrap doesn't seem safe, not all errors have a ref. We can easily compare without unwrap